### PR TITLE
fix safe mode

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve? 
Store safe-mode struct in sync to avoid depending on the safe-mode struct in event context.
Set safe-mode value by the current snapshot safe-mode value .

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

Side effects

Related changes

### CODE REVIEW

**Review by lance6716 — APPROVED** [review]

**Review by lichunzhu — APPROVED** [review]
LGTM

File: syncer/syncer.go [hunk 1]
```diff
@@ -154,6 +154,7 @@ type Syncer struct {
 		sync.RWMutex
 		t time.Time
 	}
+	safeMode *sm.SafeMode
```
Comment (lichunzhu) [inline syncer/syncer.go:157]: Can we add a comment here to explain why this variable must be a local variable?

Comment (glorv) [reply syncer/syncer.go:157]: So I think we can add a comment for the `safeMode` in eventContext?

Comment (lichunzhu) [reply syncer/syncer.go:157]: That's also fine. To avoid this line is changed in the future.

**Pull request merged**